client/src/app/project-registration/autofill-project-form/autofill-project-form.component.ts

```diff
@@ -0,0 +1,75 @@
+import {Component, OnInit} from '@angular/core';
+import {FormControl, Validators} from '@angular/forms';
+import {ActivatedRoute, Router} from '@angular/router';
+import {AlertService} from '../../shared/services/alert.service';
+import {Identifier} from '../models/europe-pmc-search';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+import {IngestService} from '../../shared/services/ingest.service';
+
+@Component({
+  selector: 'app-doi-name-field',
+  templateUrl: './autofill-project-form.component.html',
+  styleUrls: ['./autofill-project-form.component.css']
+})
+
+
+export class AutofillProjectFormComponent implements OnInit {
+  publicationDoiCtrl: FormControl;
+
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private ingestService: IngestService,
+              private alertService: AlertService
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.publicationDoiCtrl = new FormControl('', Validators.required);
+  }
+
+  showError(control: FormControl): string {
+    if (control.touched && control.errors) {
+      const errors = control.errors;
+
+      if (errors['required']) {
+        return 'This field is required';
+      }
+    }
+  }
+
+  submitForm() {
+    if (this.publicationDoiCtrl.invalid) {
+      this.publicationDoiCtrl.markAsTouched();
+    } else {
+      if (this.publicationDoiCtrl.value) {
+        const doi = this.publicationDoiCtrl.value;
+        this.doesDoiExist(doi).subscribe(doiExists => {
+            if (doiExists) {
```

Comment:
yeps, I plan on using that in the second iteration, especially as Gabs probably wants in-line errors, similar to the `required` error, instead of banner based errors, like the ones used right now.

I'll leave this conversation as unresolved so that I can refer to it in the second pull request